### PR TITLE
feat(nuxt): wrap `#components` client exports with createClientOnly

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolve https://github.com/nuxt/framework/issues/7085 , related https://github.com/nuxt/framework/issues/7409

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

hi ! i'm not sure if this should be considered as  bug fix or feature. This PR allow importing `.client` component as client only components when using explicit imports from `#components`. It also adds `ServerPlaceholder` as the `.client` server-side fallback   if there's no `.server` component to avoid hydration issues.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by huang-julien** [comment]
convert to draft due to lazy loaded client components not being lazy loaded

**Comment by huang-julien** [comment]
@danielroe
Importing from "#components" automatically break the tests due to `BreakServer` being exported in `components.server.mjs`.
It happens also on the edge channel. 
Do you have any tips or solution to test this ? Maybe renaming `BreakServer` to `BreakServer.client` ? this shouldn't break the test of `/client` since the import is done with it's path.

**Comment by danielroe** [comment]
I'd be happy with renaming `BreaksServer` :+1: Would you confirm that this still preserve the integrity of the test? (That is, that if you set `experimental.treeshakeClientOnly` to false, the test breaks.)

**Comment by huang-julien** [comment]
setting treeshakeClientOnly to false does not break the tests :+1:

File: packages/nuxt/src/components/scan.ts [hunk 1]
```diff
@@ -130,5 +131,17 @@ export async function scanComponents (dirs: ComponentsDir[], srcDir: string): Pr
...
+  components.forEach((component) => {
+    if (component.mode === 'client' && !components.some(c => c.pascalName === component.pascalName && c.mode === 'server')) {
+      components.push({
+        ...component,
+        mode: 'server',
+        filePath: resolve(distDir, 'app/components/server-placeholder'),
+        chunkName: 'components/' + component.kebabName
+      })
+    }
+  })
```
Comment (danielroe) [inline packages/nuxt/src/components/scan.ts:144]: this should probably be located after the `components:extend` hook instead:

https://github.com/nuxt/framework/blob/eab4706614199594006b2132f57e838718ac3e73/packages/nuxt/src/components/module.ts#L146-L150

**Review by danielroe — APPROVED** [review]

**Pull request merged**